File: packages/cli/src/ckpt/commands/status.py
"""ckpt status — git status with staging reasoning info."""

from __future__ import annotations

import json
import subprocess
import sys

import click

from ckpt.core import git
from ckpt.storage.local import LocalStore


@click.command(
    context_settings={"ignore_unknown_options": True, "allow_extra_args": True},
)
@click.pass_context
def status(ctx: click.Context):
    """Show working tree status with pending reasoning."""
    git_args = ["status", *ctx.args]
    try:
        git.run(*git_args)
    except subprocess.CalledProcessError as e:
        sys.exit(e.returncode)

    # Show pending staging records with reasoning
    try:
        store = LocalStore()
        records = store.get_staging_records()
        store.close()
    except Exception:
        # Storage not initialized yet — that's fine
        return

    if records:
        click.echo("\n--- ckpt staging ---")
        for r in records:
            files = json.loads(r["files"]) if isinstance(r["files"], str) else r["files"]
            reasoning = r.get("reasoning") or "(no reasoning)"
            click.echo(f"  {reasoning}")
            for f in files:
                click.echo(f"    {f}")

File: packages/cli/src/commands/status.ts
import { existsSync } from 'fs';
import { Command } from 'commander';
import { run } from '../core/git.js';
import { getCkptDir } from '../core/config.js';
import { LocalStore } from '../storage/local.js';

export const status = new Command('status')
  .description('Show git status with pending reasoning')
  .allowUnknownOption(true)
  .action((_opts: unknown, cmd: Command) => {
    const result = run(['status', ...(cmd.args ?? [])]);

    if (existsSync(getCkptDir())) {
      const store = new LocalStore();
      const records = store.getStagingRecords();

      if (records.length > 0) {
        console.log('\n--- ckpt staging ---');
        for (const record of records) {
          const reasoning = record.reasoning as string | null;
          console.log(`  ${reasoning ?? '(no reasoning)'}`);
          const files = JSON.parse((record.files as string) ?? '[]') as string[];
          for (const f of files) {
            console.log(`    ${f}`);
          }
        }
      }

      store.close();
    }

    process.exit(result.exitCode);
  });
